backend/layers/business/business.py

```diff
@@ -221,13 +220,17 @@ def update_collection_version(self, version_id: CollectionVersionId, body: Colle
         else:
             new_doi = next((link.uri for link in body.links if link.type == "DOI"), None)
 
+        # Determine if publisher metadata should be unset, ignored or set at the end of the method.
+        # Note: the update needs to be done at the end to ensure atomicity
+        unset_publisher_metadata = False
+        set_publisher_metadata = None
+
         if old_doi and new_doi is None:
             # If the DOI was deleted, remove the publisher_metadata field
-            self.database_provider.save_collection_publisher_metadata(version_id, None)
+            unset_publisher_metadata = True
         elif (new_doi is not None) and new_doi != old_doi:
             # If the DOI has changed, fetch and update the metadata
-            publisher_metadata = self._get_publisher_metadata(new_doi, errors)
-            self.database_provider.save_collection_publisher_metadata(version_id, publisher_metadata)
+            set_publisher_metadata = self._get_publisher_metadata(new_doi, errors)
```

Comment:
nit: set_publisher_metadata sounds more like a flag name next to the unset_publisher_metadata flag. Maybe this can be updated_publisher_metadata instead? But I don't feel super strongly, just an idea

Reply:
I went for `publisher_metadata_to_set`. Let me know if that makes sense